### PR TITLE
Site Editor Domain upsell: Remove callout on site-editor

### PR DESCRIPTION
**Pull request by lupus2k**
As mentioned on Slack conversation, the Site Editor Domain Upsell was showing on the Site editor and it covers a dropdown.
Slack: p1675964776091489-slack-CKZHG0QCR

| With Upsell | Without Upsell |
| --- | --- |
| image | image |

## Proposed Changes

Filter site-editor to not show the Domain Upsell.

## Testing Instructions

- Use a site with a Free plan and without custom domains.
- Go `Appearance`, click on `Editor (Beta)`
- You should not see the Domain Upsell callout

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] ~~[Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?~~
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] ~~Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)~~
- [ ] ~~Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?~~
- [ ] ~~For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?~~

### CODE REVIEW

File: client/gutenberg/editor/calypsoify-iframe.tsx [hunk 1]
```diff
@@ -713,7 +717,7 @@ class CalypsoifyIframe extends Component< ComponentProps, State > {
 				{ /* eslint-disable-next-line wpcalypso/jsx-classname-namespace */ }
 				<div className="main main-column calypsoify is-iframe" role="main">
 					{ ! isIframeLoaded && <Placeholder /> }
-					{ isIframeLoaded && (
+					{ isIframeLoaded && this.isEditorTypePost() && (
```
Comment (Copons) [inline client/gutenberg/editor/calypsoify-iframe.tsx]: IMHO we could simplify this by just checking for the Site Editor:

```suggestion
					{ isIframeLoaded && this.props.editorType !== 'site' && (
```

Comment (lupus2k) [reply client/gutenberg/editor/calypsoify-iframe.tsx]: You're right, I updated the PR

**Review by DustyReagan — APPROVED** [review]
Domain upsell is no longer showing on the editor, but still showing on posts preview. 👍

**Comment by Copons** [comment]
Thank you for this quick fix! 🙇‍♂️

**Pull request merged**